### ISSUE TITLE
Rich Text field ignores custom Label component

### ISSUE THREAD

**Issue by CherryDT** [payloadcms/payload#6081]
### Describe the Bug

I have a rich text field (using Slate) which has a custom `Label` component. The component is ignored and the default label is rendered. The same component works in other fields.

Looking at the `admin.components` object at runtime, it is suddenly empty:

![image](https://github.com/payloadcms/payload/assets/6010463/8a776638-0f9c-4e00-a870-aab1bf050514)

Probably something goes wrong while merging the data in `admin` because I can see both `elements`/`leaves` (which comes from the global editor config) and `disableBulkEdit` (which comes from the field config), so there must be some kind of merging happening, and I assume my field config's `components` object gets overridden somehow.

### To Reproduce

Create a Slate rich text field with a custom label component in the field config.

### Payload Version

2.14.2

### Adapters and Plugins

db-mongodb, bundler-webpack, richtext-slate

**Comment by JessChowdhury** [comment]
@CherryDT can you show me where in your config you are setting this custom label?

```ts
    {
      name: 'richTextSlate',
      type: 'richText',
      admin: {
        components: {
          Label: () => 'Test Custom Label', // Here?
        },
      },
      editor: slateEditor({...}),
    },
```
Like this? Or within the `slateEditor` adapter?

A reproduction would be helpful here, I have not been able to replicate this issue.

**Comment by CherryDT** [comment]
Yes, at the point you showed. It works fine there for other fields. It does work when I set it within the `slateEditor` adapter, I just noticed.

Note that I don't set an `editor` at the field. I set it globally.

**Comment by JessChowdhury** [comment]
I have tested again with the `editor` being defined globally - the label component still works as expected in both cases. 

It's possible that something else in your code is affecting your ability to over-write the label component. Please try to replicate this issue in a basic environment and I can take another look - for now I'm going to convert this to a discussion.
